src/axolotl/utils/quantization.py

```diff
@@ -45,50 +53,59 @@ def get_ptq_config(
             or if the group size is not specified for int8 or int4 weight only quantization.
     """
     if activation_dtype is None:
-        if not weight_dtype.value.is_signed:  # type: ignore[attr-defined,union-attr]
-            return UIntXWeightOnlyConfig(
-                dtype=weight_dtype.value,
-                group_size=group_size,
-                set_inductor_config=False,
-            )
-        if weight_dtype == TorchIntDType.int8:
-            if group_size is None:
-                raise ValueError(
-                    "group_size must be specified for int8 weight only quantization"
-                )
-            return Int8WeightOnlyConfig(
-                group_size=group_size,
-            )
-        if weight_dtype == TorchIntDType.int4:
-            if group_size is None:
-                raise ValueError(
-                    "group_size must be specified for int4 weight only quantization"
-                )
-            return Int4WeightOnlyConfig(
-                group_size=group_size,
-            )
-    if activation_dtype == TorchIntDType.int4 and weight_dtype == TorchIntDType.int4:
-        return Int4DynamicActivationInt4WeightConfig()
-    if activation_dtype == TorchIntDType.int8 and weight_dtype == TorchIntDType.int8:
-        return Int8DynamicActivationInt8WeightConfig()
-    if activation_dtype == TorchIntDType.int8 and weight_dtype == TorchIntDType.int4:
-        return Int8DynamicActivationInt4WeightConfig()
+        if weight_dtype == TorchAOQuantDType.int8:
+            raise ValueError("Int8WeightOnlyConfig is not supported by torchao QAT.")
+        if weight_dtype == TorchAOQuantDType.int4:
+            return Int4WeightOnlyConfig(group_size=group_size or -1, version=2)
```

Comment:
note that if you want BC you should use version=1 (that was the one you had before this PR, which uses the tinygemm kernel). If you want to default to the fbgemm kernel, then you should use version 2